README.mkd

```diff
@@ -242,6 +242,58 @@ As of November 2021 the additional tail-call entry points are only
 implemented in compiler-rt, and calls will only be generated by LLVM
 when the option `-mllvm -save-restore-tailcall` is specified.
 
+## Conventions for vendor extension
+
+Support for custom instruction set extensions are an important part of RISC-V,
+with large encoding spaces reserved of vendor extensions.
+
+However, there are no official guidelines on how to use this encoding space or
+on naming the mnemonics. This section defines guidelines which vendors are
+expected to follow if upstreaming support for their extensions. Although
+vendor-provided toolchains are free to make different choices, they are strongly
+urged to align with these guidelines in order to ensure there is a straightforward
+path for upstreaming in the future.
+
+### Vendor extension naming scheme
+
+According to the RISC-V ISA spec, non-standard extensions are named using a single `x`
+followed by an alphabetical name and an optional version number.
+
+To make it easier to identify and prevent naming conflict, vendor extensions must start
+with a vendor name, which could be an abbreviation of the full name.
+
+For example:
+- `XVentanaCondOps` from Ventana
+- `Xsfcflushdlone` from SiFive
+- `Xtheade` from T-head
```

Comment:
Can we drop the Xtheade example for now, as the names of the T-Head extensions may still change?

We are working to further subdivide these into multiple orthogonal extensions with T-Head (e.g. CMOs, instructions shadowing Zbb, instructions shadowing Zba, multiply-accumulates, etc.)...

Reply:
Removed.